misc tweaks on golang docs (#14)
ssbc/dev.scuttlebutt.nz: docs/golang/README.md

The `plugins2/names` package implements the same muxrpc interface as [ssb-names](https://github.com/ssbc/ssb-names). It does a similar thing to `graph` but for `type:about` messages and the name, image and description information in them.

### [`margaret/multilog`](https://pkg.go.dev/go.cryptoscope.co/margaret)

We wanted to have a unified way to access different kinds of sublogs, that is, logs-as-slices into a larger log. One example: accessing parts of the big receive log (the posts by one particular feed, for instance).

Enter [MultiLog](https://pkg.go.dev/go.cryptoscope.co/margaret/multilog), which returns `margaret.Log`s for specific addresses.

Package `multilog` is one of a few ways that go-ssb creates an overarching structure (indexes and views) of the
messages being received from all over the scuttleverse. Another prominent indexing solution is
the [`margaret/indexes`](https://godocs.io/go.cryptoscope.co/margaret/indexes) package, which
more or less functions as a key-value store. One of the differences between `margaret/indexes` and
`margaret/multilog`: the former stores arbitrary data (think: strings of your choosing), while

> the latter only stores ssb messages.

Reviewer: ```suggestion
the latter only stores (references to) ssb messages.
```

#### A `multilog` explanation

One way to think about the `multilog` package: 

A multilog can be seen as a _[tree](https://en.wikipedia.org/wiki/Tree_(data_structure))-like_
index. The multilog itself is a log—or the root of the tree—which leads to other logs. These
other logs are called _sublogs_, and each sublog has many leaves. Each leaf corresponds to some

> message. What type of message is stored depends on the particular type of data being indexed by the multilog in question (all follow messages by `<peer>`, for example).

Reviewer: ```suggestion
message. What type of message is stored depends on the particular type of data being indexed by the multilog in question (i.e. all `type:contact` messages, all messages by `<peer>`, etc.).
```

Reviewer: the example is not directly indexed but could be constructed by boolean combination of two indexes.
